kratos/utilities/function_parser_utility.cpp

```diff
@@ -36,7 +60,7 @@ BasicGenericFunctionUtility::BasicGenericFunctionUtility(const std::string& rFun
     InitializeParser();
 
     // Check if it depends on space
-    if (mFunctionBody.find(std::string("x")) == std::string::npos &&
+    if (CheckThereIsNotx(mFunctionBody)                           &&
         mFunctionBody.find(std::string("y")) == std::string::npos &&
         mFunctionBody.find(std::string("z")) == std::string::npos &&
         mFunctionBody.find(std::string("X")) == std::string::npos &&
```

Comment:
How do you check that is only x?, it is the same as saying not surrounded by other characters. If you find a character, you don't know if it is part of a keyword

Reply:
Depends if we allow e.g. `xy` or enforce ´x*y`
not saying I am correct, just giving my thoughts

Reply:
IMO this is the easiest way with the least coerner cases

Reply:
`std::regex_search(expr, std::regex("\\b[xyz]\\b"))`?

Reply:
Good idea but does not work, I am not experienced in regex enough to do the correct reguar expression

Reply:
![image](https://user-images.githubusercontent.com/19991680/154946573-ae964fd1-4ee4-4f6d-900d-e00deda46616.png)

Reply:
Fixed with  regex like a pro, thanks @oberbichler 

![image](https://user-images.githubusercontent.com/19991680/154950721-0187842e-0b70-4cf5-8f38-4e47f17e6658.png)

Today I learned regex

Reply:
`\b...\b` is similar to "Match Whole Word" in VS Code search function. `[xyz]` matches any of the chars. There is also an option for case-insensitivity. Just wondering why [`\b[xyzXYZ]\b` is not equivalent to "depends on space"](https://regex101.com/r/SzNK98/1) and [`\bt\b` to "depends on time"](https://regex101.com/r/T9TIny/1). The xy=x*y thing?

[...and keep in mind ;)](http://regex.info/blog/2006-09-15/247)

Reply:
With the regex looks like it fails, do not know why

Reply:
I reverted the regex, I am unable to translate to C++ a proper regex code to work:  https://regex101.com/r/YYDMKo/1 , this works, but once I translate to C++ it does not work

Reply:
This is quite strange because the same code seems to compile just fine in isolation:

https://godbolt.org/z/r993dbqxe

Reply:
The problem is not that, the problem is that "x" and "2*x" does not work, so you need an additional check, that in C++ fails, but in JS not

Reply:
I thought that it might be enough to search for `xyz` or `t` (instead of checking for non-"exp"/non-"sqrt")

https://godbolt.org/z/hc41PYr7P

but I am probably missing some corner cases(?)

Reply:
I tried that the other day, and it didn't work, I will try again

Reply:
According to this should work now: https://regex101.com/r/upucEk/1